### PR TITLE
upgrade repo infra to f85734f673056977d8ba04b0386394b684ca2acb

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind failing-test
> /kind feature
> /kind flake

/priority critical-urgent

**What this PR does / why we need it**: picks up fixes in https://github.com/kubernetes/repo-infra/pull/120/files

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig testing
/sig release

### CODE REVIEW

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -37,9 +37,9 @@ versions.check(minimum_bazel_version = "0.23.0")
 
 http_archive(
     name = "io_k8s_repo_infra",
-    sha256 = "4a8384320fba401cbf21fef177aa113ed8fe35952ace98e00b796cac87ae7868",
-    strip_prefix = "repo-infra-df02ded38f9506e5bbcbf21702034b4fef815f2f",
-    urls = mirror("https://github.com/kubernetes/repo-infra/archive/df02ded38f9506e5bbcbf21702034b4fef815f2f.tar.gz"),
+    sha256 = "441631e653b281e2b71a2b1d1b69d7bcc827e08b81e7ae709134ae9f4c2afed9",
```
Comment (BenTheElder) [inline build/root/WORKSPACE]: grabbed with curl | sha256sum, hopefully correct!

Comment (cblecker) [reply build/root/WORKSPACE]:
```suggestion
    sha256 = "4a8384320fba401cbf21fef177aa113ed8fe35952ace98e00b796cac87ae7868",
```

Comment (BenTheElder) [reply build/root/WORKSPACE]: nope, corrected it.

**Comment by BenTheElder** [comment]
/priority important-longterm
we should still upgrade repo-infra

**Comment by cblecker** [comment]
Everything here looks fine with me: https://github.com/kubernetes/repo-infra/compare/df02ded38f9506e5bbcbf21702034b4fef815f2f...f85734f673056977d8ba04b0386394b684ca2acb

/lgtm
/approve

**Comment by BenTheElder** [comment]
There's still something wrong with the python runfiles and the gcs_upload
implementation in repo-infra but it only affects up to date bazel, not
0.23.2

**Pull request merged**